rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -977,14 +978,21 @@ protected Web3InformationRetriever getWeb3InformationRetriever() {
     }
 
     protected ReceiptStore buildReceiptStore() {
-        int receiptsCacheSize = getRskSystemProperties().getReceiptsCacheSize();
-        KeyValueDataSource ds = LevelDbDataSource.makeDataSource(Paths.get(getRskSystemProperties().databaseDir(), "receipts"));
+        RskSystemProperties rskSystemProperties = getRskSystemProperties();
+        int receiptsCacheSize = rskSystemProperties.getReceiptsCacheSize();
+        KeyValueDataSource ds = LevelDbDataSource.makeDataSource(Paths.get(rskSystemProperties.databaseDir(), "receipts"));
 
         if (receiptsCacheSize != 0) {
             ds = new DataSourceWithCache(ds, receiptsCacheSize);
         }
 
-        return new ReceiptStoreImpl(ds);
+        //noinspection SwitchStatementWithTooFewBranches
+        switch (rskSystemProperties.receiptsDataFormatVersion()) {
```

Comment:
I'm not sure to have this feature as a configuration setting. 

We could use V2 directly in the code. Then, having the fallback behavior, once the new code is distributed, the receipts database will be have the new version, automatically. And no one can set the setting to v1. If it were changed, the database will not work.

On the other hand, the settings is good for testing and comparison using the same code base in the rest of the application

Reply:
removed that feature flag